### PR TITLE
Zero all malleable fields before execution

### LINKED ISSUE #542: Zero all malleable fields before execution

### PR DESCRIPTION
**Pull request by Dentosal**
Closes #542.

Also changes `tx.receiptsRoot` to only be updated after the tx has terminated, as the intermediate values are no longer useful. The speeds up and simplifies implementation of receipt-pushing opcodes.

### CODE REVIEW

**Comment by xgreenx** [comment]
Maybe it makes sense to remove the getter for malleable fields since they are zero all the time. We don't need to change the discriminant for values, we can leave gaps

**Comment by xgreenx** [comment]
Before merging this PR, I would suggest doing https://github.com/FuelLabs/fuel-core/issues/1620 first. It will help identify all problematic fields. Maybe we missed something here

**Comment by Dentosal** [comment]
CI not passing due to https://github.com/FuelLabs/fuel-specs/issues/547

File: src/fuel-vm/instruction-set.md [hunk 1]
```diff
@@ -2491,19 +2490,13 @@ Get [fields from the transaction](../tx-format/transaction.md).
 | `GTF_INPUT_COIN_OWNER`                    | `0x203` | Memory address of `tx.inputs[$rB].owner`                          |
 | `GTF_INPUT_COIN_AMOUNT`                   | `0x204` | `tx.inputs[$rB].amount`                                           |
 | `GTF_INPUT_COIN_ASSET_ID`                 | `0x205` | Memory address of `tx.inputs[$rB].asset_id`                       |
-| `GTF_INPUT_COIN_TX_POINTER`               | `0x206` | Memory address of `tx.inputs[$rB].txPointer`                      |
```
Comment (Voxelot) [inline src/fuel-vm/instruction-set.md:2494]: weren't we going to use tx_pointer as a replacement for the maturity field? Why are we not setting it anymore?

cc @xgreenx

Comment (Voxelot) [reply src/fuel-vm/instruction-set.md:2494]: Discussed on a call - the tx_pointer will be unreliable for predicates to use as a maturity check. The most sensible approach for now is to have predicates check the tx-wide maturity policy.

**Review by Voxelot — APPROVED** [review]

**Comment by Voxelot** [comment]
For future reference, removing the intermediate receipt roots is possible now that we are going with a zk-hybrid proving approach.

**Review by xgreenx — APPROVED** [review]

**Pull request merged**